core/src/main/java/com/netease/arctic/utils/map/RocksDBBackend.java

```diff
@@ -47,37 +45,65 @@
 import java.util.UUID;
 import java.util.stream.Collectors;
 
-public class RocksDBBackend {
+public class RocksDBBackend<K, T> {
   private static final Logger LOG = LoggerFactory.getLogger(RocksDBBackend.class);
   private static final String BACKEND_BASE_DIR = System.getProperty("rocksdb.dir");
   private static final ThreadLocal<RocksDBBackend> instance =
-          new ThreadLocal<>().withInitial(() -> create());
+          new ThreadLocal<>();
 
   public static RocksDBBackend getOrCreateInstance() {
     RocksDBBackend backend = instance.get();
+    if (backend == null) {
+      backend = create();
+    }
     if (backend.closed) {
       backend = create();
       instance.set(backend);
     }
     return backend;
   }
 
+  public static <K, T> RocksDBBackend<K, T> getOrCreateInstance(Serializer<K> keySerializer,
+      Serializer<T> valueSerializer) {
+    RocksDBBackend backend = instance.get();
+    if (backend == null) {
+      backend = create(keySerializer, valueSerializer);
+    }
+    if (backend.closed) {
+      backend = create(keySerializer, valueSerializer);
+      instance.set(backend);
+    }
+    return backend;
+  }
+
   private Map<String, ColumnFamilyHandle> handlesMap = new HashMap<>();
   private Map<String, ColumnFamilyDescriptor> descriptorMap = new HashMap<>();
   private RocksDB rocksDB;
   private boolean closed = false;
   private final String rocksDBBasePath;
   private long totalBytesWritten;
+  private Serializer<K> keySerializer;
+  private Serializer<T> valueSerializer;
 
```

Comment:
we could put Serializers into Map implementations and use byte[] for put and get